src/firmware_new/boards/frankie_v1/tim.c

```diff
@@ -0,0 +1,146 @@
+/**
+ ******************************************************************************
+ * File Name          : TIM.c
+ * Description        : This file provides code for the configuration
+ *                      of the TIM instances.
+ ******************************************************************************
+ * @attention
+ *
+ * <h2><center>&copy; Copyright (c) 2020 STMicroelectronics.
+ * All rights reserved.</center></h2>
+ *
+ * This software component is licensed by ST under Ultimate Liberty license
+ * SLA0044, the "License"; You may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at:
+ *                             www.st.com/SLA0044
+ *
+ ******************************************************************************
+ */
+
+/* Includes ------------------------------------------------------------------*/
+#include "tim.h"
+
+/* USER CODE BEGIN 0 */
+
+/* USER CODE END 0 */
+
+TIM_HandleTypeDef htim4;
+
+/* TIM4 init function */
+void MX_TIM4_Init(void)
+{
+    TIM_MasterConfigTypeDef sMasterConfig = {0};
+    TIM_OC_InitTypeDef sConfigOC          = {0};
+
+    htim4.Instance               = TIM4;
+    htim4.Init.Prescaler         = 9 - 1;
```

Comment:
What is the reason for subtracting? Same with `Init.Period`

Reply:
This is all autogenerated,

My guess is maybe the setting in cube is 9 but the value needs to be one less? But they don't just generate 8 because it would look like the generation is faulty, so they generated 9-1, instead of 8 to be explicit

But I honestly have no idea why its not 8 but 9 - 1.

Reply:
lmao